### PR TITLE
Remove production use of DatabaseConfigPersistence

### PR DESCRIPTION
**Pull request by cgardens**
Depends on PR to add new db base test calss: https://github.com/airbytehq/airbyte/pull/19640

## What
* Removes `DatabaseConfigPersistence`!!! 

## How
* It does this by moving relevant database queries into the `ConfigRepository`
* Took this opportunity to start to split tests into their own classes by resource (e.g. workspace, actor definition). I did _not_ break resources into their own persistences. That's too much to do in one PR, though we should do that next. I did the tests, because otherwise I couldn't wade through what was going on safely enough.
* Removes test classes that were just testing `ConfigPersistence`. Moved any relevant business logic tests that were hidden in there (which there were a few, which is kinda scary because those tests were testing the old `ConfigPersistence` implementations and not the ones that are currently used) and moved them into appropriate classes.
* Even with all the code that is just being moved, we are still managing to drop nearly 2K lines of code. 😄 

## Future Work
* `ConfigRepositoryE2EReadWriteTest.java` - this class is a monster. Decided not to tackle it now, but as we split persistences and their tests by resource, we should pick at this. It basically has tests for CRUD operations for every single resource.
* Split out test suites for workspace, actor definitions, and health checks. It should make it relatively straight forward to write separate persistences for them as a follow up.

## Recommended reading order
1. `ConfigRepository.java`
2. `ActorDefinitionMigratorTest.java` -- this is the one where there were a lot of test that got move to it from ConfigPersistence. 
3. `WorkspacePeristenceTest.java`
4. `ActorDefinitionPersistenceTest.java`
5. `HealthCheckPersistenceTest.java`
6. the rest...

## 🚨 User Impact 🚨
This PR does move around a lot of the queries from the `ConfigPersistence` to the `ConfigRepository`. There is a lot of testing but it's not that methodical or perfect coverage. There is some risk here that there is some mistake in a query that isn't caught by testing.

### CODE REVIEW

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/BaseDatabaseConfigPersistenceTest.java [hunk 1]
```diff
@@ -1,231 +0,0 @@
-/*
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/BaseDatabaseConfigPersistenceTest.java:1]: Rebuilt this class as `BaseConfigDatabaseTest`.

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ConfigRepositoryTest.java [hunk 2]
```diff
@@ -1,511 +0,0 @@
-/*
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ConfigRepositoryTest.java:1]: all tests in this class were moved into separate `*PersistenceTest` (`HealthCheckPersistenceTest.java` `WorkspacePersistenceTest.java`, and `ActorDefinitionPersistenceTest.java`)

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/DatabaseConfigPersistenceUpdateConnectorDefinitionsTest.java [hunk 3]
```diff
@@ -1,230 +0,0 @@
...
-import java.util.stream.Collectors;
-import org.jooq.SQLDialect;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Test;
-
-/**
- * Unit test for the {@link DatabaseConfigPersistence#updateConnectorDefinitions} method.
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/DatabaseConfigPersistenceUpdateConnectorDefinitionsTest.java:39]: this code path wasn't actually used anymore, so these tests were for a dead code path.

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/StatePersistenceTest.java [hunk 4]
```diff
@@ -17,33 +17,27 @@
...
 import org.jooq.JSONB;
-import org.jooq.SQLDialect;
 import org.jooq.impl.DSL;
-import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-class StatePersistenceTest extends BaseDatabaseConfigPersistenceTest {
+class StatePersistenceTest extends BaseConfigDatabaseTest {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/StatePersistenceTest.java:40]: Changes in this class:
* Remove unneeded db setup, let `BaseConfigDatabaseTest`
* Remove usage of `dslContext` and replace it with `database`.

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/StreamResetPersistenceTest.java [hunk 5]
```diff
@@ -8,49 +8,25 @@
...
 import java.util.UUID;
-import org.jooq.SQLDialect;
-import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class StreamResetPersistenceTest extends BaseDatabaseConfigPersistenceTest {
+class StreamResetPersistenceTest extends BaseConfigDatabaseTest {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/StreamResetPersistenceTest.java:20]: Changes in this class:
* Remove unneeded db setup, let `BaseConfigDatabaseTest`
* Remove usage of `dslContext` and replace it with `database`.

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ActorDefinitionPersistenceTest.java [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class ActorDefinitionPersistenceTest extends BaseConfigDatabaseTest {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ActorDefinitionPersistenceTest.java:32]: lift and shift of tests from `ConfigRepository`

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspacePersistenceTest.java [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+@SuppressWarnings({"PMD.LongVariable", "PMD.AvoidInstantiatingObjectsInLoops"})
+class WorkspacePersistenceTest extends BaseConfigDatabaseTest {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspacePersistenceTest.java:30]: Changes in this file:
* lift and shift of tests from `ConfigRepository`.
* replace `ConfigPersistence` with `ConfigRepository`.
* intentionally not changing any of the tests themselves.

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/HealthCheckPersistenceTest.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.airbyte.db.Database;
+import java.sql.SQLException;
+import org.jooq.Result;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class HealthCheckPersistenceTest {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/HealthCheckPersistenceTest.java:19]: Changes in this file:
* lift and shift of tests from `ConfigRepository`.
* intentionally not changing any of the tests themselves.

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StandardSyncPersistence.java [hunk 9]
```diff
@@ -117,6 +119,11 @@ public void clearUnsupportedProtocolVersionFlag(final UUID actorDefinitionId,
     });
   }
 
+  public List<StreamDescriptor> getAllStreamsForConnection(final UUID connectionId) throws ConfigNotFoundException, IOException {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StandardSyncPersistence.java:122]: moved from `ConfigRepository`

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ConfigRepositoryE2EReadWriteTest.java [hunk 10]
```diff
@@ -609,4 +610,37 @@ private void insertCatalogFetchEvent(final DSLContext ctx, final UUID sourceId,
         .execute();
   }
 
+  private static void writeActorCatalog(final Database database, final List<ActorCatalog> configs) throws IOException, SQLException {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ConfigRepositoryE2EReadWriteTest.java:613]: moved helpers from `DatabaseConfigPersistence` that were only used in these tests.

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ActorDefinitionMigratorTest.java [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.jooq.Record1;
+import org.jooq.Result;
+import org.jooq.Table;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ActorDefinitionMigratorTest extends BaseConfigDatabaseTest {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ActorDefinitionMigratorTest.java:41]: Add tests from `DatabaseConfigPersistenceTest` that were specific to migrator.

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java [hunk 12]
```diff
@@ -237,18 +282,23 @@ public StandardWorkspace getStandardWorkspaceFromConnection(final UUID connectio
...
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  private Stream<StandardSourceDefinition> sourceDefQuery(final Optional<UUID> sourceDefId, final boolean includeTombstone) throws IOException {
+    return database.query(ctx -> ctx.select(ACTOR_DEFINITION.asterisk())
+        .from(ACTOR_DEFINITION)
+        .where(ACTOR_DEFINITION.ACTOR_TYPE.eq(ActorType.source))
+        .and(sourceDefId.map(ACTOR_DEFINITION.ID::eq).orElse(noCondition()))
+        .and(includeTombstone ? noCondition() : ACTOR_DEFINITION.TOMBSTONE.notEqual(true))
+        .fetchStream())
+        .map(DbConverter::buildStandardSourceDefinition)
+        // todo(cgardens) - the impls for default protocol version were different in list and get, which is
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java:298]: @gosusnp we have 2 different implementations of this default protocol version behavior. which is right?
1. https://github.com/airbytehq/airbyte/blob/master/airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java#L204-L208
2. https://github.com/airbytehq/airbyte/blob/master/airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java#L244-L245

File: airbyte-server/src/main/java/io/airbyte/server/handlers/ConnectionsHandler.java [hunk 13]
```diff
@@ -179,7 +179,7 @@ public ConnectionRead createConnection(final ConnectionCreate connectionCreate)
       eventRunner.createConnectionManagerWorkflow(connectionId);
     } catch (final Exception e) {
       LOGGER.error("Start of the connection manager workflow failed", e);
-      configRepository.deleteStandardSyncDefinition(standardSync.getConnectionId());
```
Comment (cgardens) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/ConnectionsHandler.java:182]: this name was wrong and misleading. i couldn't help myself.

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ActorDefinitionMigratorTest.java [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ActorDefinitionMigratorTest extends BaseConfigDatabaseTest {
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/ActorDefinitionMigratorTest.java:31]: Add tests from DatabaseConfigPersistenceTest that were specific to migrator.

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java (hunk 12 above)
Comment (gosusnp) [reply airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java:294]: Good catch, the get is the correct behavior, we should always use the default if missing.

**Comment by cgardens** [comment]
I would really like to get a code review on this to remove the `DatabaseConfigPersistence` before we break for Thanksgiving, so I can merge it right when we get back and we can be rid of the damn thing! :knife: :knife: :knife: :smiling_imp: :turkey: 

@benmoriceau @gosusnp @jdpgrailsdev lmk if there's anything I can do to aid the review.

**Review by benmoriceau — APPROVED** [review]

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Pull request merged**